engine/access/ingestion/engine.go

```diff
@@ -544,8 +544,16 @@ func (e *Engine) trackExecutedMetricForCollection(light *flow.LightCollection) {
 	}
 }
 
-// handleCollection handles the response of the a collection request made earlier when a block was received
-func (e *Engine) handleCollection(_ flow.Identifier, entity flow.Entity) error {
+type ExecutionMetricForCollection func(light *flow.LightCollection)
+
+// HandleCollection handles the response of the a collection request made earlier when a block was received
+func HandleCollection(
```

Comment:
if this moves to `indexer_core`, it can take the actual collection instead of an entity, and the caller can do the metrics collection instead of passing it in here.

Reply:
I understand that after moving `HandleCollection` to `indexer_core` I can pass `flow.Collection` instead of `entity`. But what about metrics that were called in ingestion engine inside HandleCollection. I am talking about this method `trackExecutedMetricForCollection`. Should I move all missing parts for it to `indexer_core`? Also I need to remove
`collectionHandler requester.HandleFunc ` field and use `HandleCollection` straight inside `indexer_core`?
 Thanks in advance.

Reply:
> @peterargue I understand that after moving `HandleCollection` to `indexer_core` I can pass `flow.Collection` instead of `entity`. But what about metrics that were called in ingestion engine inside HandleCollection. I am talking about this method `trackExecutedMetricForCollection`. Should I move all missing parts for it to `indexer_core`? 

I think we could just call that in `handleCollection` before calling `HandleCollection`

> Also I need to remove `collectionHandler requester.HandleFunc ` field and use `HandleCollection` straight inside `indexer_core`? Thanks in advance.

yes, exactly